src/main/com/smithsmodding/armory/common/factories/ArmorFactory.java

```diff
@@ -123,24 +197,10 @@ private boolean validateOldAgainstNewAddons(@Nonnull ArrayList<IMultiComponentAr
         return continueCrafting;
     }
 
-    private boolean validateNewAgainstNewAddons(@Nonnull ArrayList<IMultiComponentArmorExtensionInformation> newAddons) {
-        boolean continueCrafting = true;
-        Iterator<IMultiComponentArmorExtensionInformation> externalIterator = newAddons.iterator();
-
-        while (continueCrafting && externalIterator.hasNext()) {
-            Iterator<IMultiComponentArmorExtensionInformation> internalIterator = newAddons.iterator();
-            IMultiComponentArmorExtension externalExtension = externalIterator.next().getExtension();
-
-            while (continueCrafting && internalIterator.hasNext()) {
-                IMultiComponentArmorExtension internalExtension = internalIterator.next().getExtension();
-                continueCrafting = externalExtension.validateCrafting(internalExtension, false);
-            }
-        }
-
-        return continueCrafting;
-    }
-
-    public ArrayList<IMultiComponentArmorExtensionInformation> compressInformation(ArrayList<IMultiComponentArmorExtensionInformation> oldExtensions, ArrayList<IMultiComponentArmorExtensionInformation> newExtensions) {
+    private List<IMultiComponentArmorExtensionInformation> compressInformation(
+      List<IMultiComponentArmorExtensionInformation> oldExtensions,
+      List<IMultiComponentArmorExtensionInformation> newExtensions)
+    {
         HashMap<IMultiComponentArmorExtension, Integer> countMap = new HashMap<>();
 
         oldExtensions.forEach((i)-> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "i" parameter (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1611)